migrator/src/main/java/com/google/phonenumbers/migrator/MetadataReader.java

```diff
@@ -0,0 +1,74 @@
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.i18n.phonenumbers.metadata.model.RangesTableSchema;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * A class used to represent a given metadata zip file. Each {@link MetadataReader} instance
+ * represents a zip file containing Csv Tables that can be imported based on specified
+ * region codes without the need to extract the whole zip file.
+ */
+public final class MetadataReader {
+  private final ZipFile metadataZipFile;
+
+  public MetadataReader(String fileLocation) throws IOException {
+    this.metadataZipFile = new ZipFile(fileLocation);
+  }
+
+  /**
+   * Returns an Optional {@link CsvTable} correlating to a given regionCode. If no ranges.csv file
+   * can be found in the {@link #metadataZipFile} for the region code, an empty Optional is returned
+   */
+  public Optional<CsvTable<RangeKey>> getCsvTable(String regionCode) throws IOException {
+    String CsvTableLocation = "metadata/" + regionCode + "/ranges.csv";
+    ZipEntry csvFile = metadataZipFile.getEntry(CsvTableLocation);
+
+    if (csvFile == null) {
+      return Optional.empty();
+    }
+    return Optional.of(CsvTable.importCsv(RangesTableSchema.SCHEMA,
+        new InputStreamReader(metadataZipFile.getInputStream(csvFile))));
+  }
+
```

Comment:
You could add a comment here about the arguments that you are expecting and what format they should be in.